src/data/markdown/translated-guides/en/02 Using k6/14 Scenarios/01 Executors/05 constant-arrival-rate.md

```diff
@@ -40,7 +39,15 @@ this executor has the following options:
 ## When to use
 
 When you want to maintain a constant number of iterations without being affected by the
-performance of the system under test.
+performance of the system under test. To reliably achieve a fixed request rate, try to keep test function simple, with preferably only a single request call,
```

Comment:
I don't think that recommending to only have 1 request is a good thing. While this will simplify calculating how many VUs you need in normal cases it :

1. in practice makes it seem like the executor can't handle anything more complicated, which is far from the truth
2. again in the case that the SUT is failing the "request should take no more than 1s" thinking is still not going to fix anything when it is taking 1m ;)

Reply:
Yeah I just moved that from the old section; it struck me as pretty much invalidating all complex uses of arrival rate. Will delete.

Reply:
[e8751a1](https://github.com/grafana/k6-docs/pull/1008/commits/e8751a19a8555e8ba79f77d2e80c51b50e476cf5)